hardware/measurement/adc_stream_rate.py

```
@@ -0,0 +1,448 @@
+
+def write_csv(path: str, r, allow_drops: bool):
+    """log形式は power-budget.md の `Sample rateとlog形式` に合わせる。
+
+    同節は `時刻[us],生ADC値` を定める。2 channelのときだけ ch 列を足す。
+    時刻は block header の micros() mark を anchor に、sample index から線形に復元する。
+    """
+    used = r["used"]
+    if r["dropped_delta"] and not allow_drops:
+        raise SystemExit(
+            "取りこぼしが %d 件ある。時刻の復元が曖昧になるため CSV を書かない。\n"
+            "承知のうえで書くなら --allow-drops を付ける（時刻は近似になる）。"
```

Ruff (0.16.1)

[warning] String contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF001)